GVFS/GVFS.Platform.Mac/MacFileSystemVirtualizer.cs

```diff
@@ -165,6 +232,67 @@ private static byte[] ToVersionIdByteArray(byte[] version)
             return bytes;
         }
 
+        private bool TryGetSymLinkObjectContents(string sha, out string contents)
+        {
+            contents = null;
+            StringBuilder objectContents = new StringBuilder();
+
+            try
+            {
+                if (!this.GitObjects.TryCopyBlobContentStream(
+                    sha,
+                    CancellationToken.None,
+                    GVFSGitObjects.RequestSource.SymLinkCreation,
+                    (stream, blobLength) =>
+                    {
+                        // TODO(Mac): Find a better solution than reading from the stream one byte at at time
+                        byte[] buffer = new byte[4096];

```

Comment:
We can't use chunks as we could cross a UTF8 character barrier

Reply:
You probably don't need to worry about chunking anyway, if the contents are here are actually a path. Just make sure you have one buffer that's large enough for a path.